### ISSUE TITLE
[Request] Variable drawer width

### ISSUE THREAD

**Issue by moderndeveloperllc** [vaadin/vaadin-app-layout#114]
Right now the drawer width [is fixed](https://github.com/vaadin/vaadin-app-layout/blob/997111c7d181cb9c44421edb7e15b926f907edba/src/vaadin-app-layout.html#L103) at 16em. For those of us with small tab names, it can create a bit too much blank space. It would be nice to be able to set that as a CSS variable. It looks like the main area is already [being offset from the drawer programmatically](https://github.com/vaadin/vaadin-app-layout/blob/997111c7d181cb9c44421edb7e15b926f907edba/src/vaadin-app-layout.html#L483).

**Comment by DiegoCardoso** [comment]
That sounds a good enhancement for the component. Thanks!

**Comment by jouni** [comment]
Yeah, would be straightforward to expose a new custom property and use that for setting the other offset property.

**Comment by brianreeve** [comment]
I agree with OP that it would be nice to have styling control over the drawer width. However, you can achieve a lot of style control by following the pattern Vaadin uses to ship components with both lumo and material themes.

First, create a `my-app-layout` directory with the following files:

**`my-app-layout-styles.js`**

For a quick and dirty solution, you can copy the contents of a theme file (I copied `node_modules/@vaadin/vaadin-app-layout/theme/lumo/vaadin-app-layout-styles.js`) and make the necessary modifications.

I heavily modified the app layout theme file for my app, but to set the drawer width, I have:

**`app-layout.js`**

```javascript
import './my-app-layout-styles.js';
import '../../node_modules/@vaadin/vaadin-app-layout/src/vaadin-app-layout.js';
```

Then, just import `my-app-layout/app-layout.js` instead of one in `node_modules`.

**Comment by trojanflash** [comment]
Easy solution for change drawer width:
Create some css `frontend/styles/my-app-layout-styles.css` with 
`[part="drawer"] {
    width: 8em  !important;
}`

add css import
`@CssImport(value = "./styles/my-app-layout-styles.css", themeFor = "vaadin-app-layout")`

**Comment by anezthes** [comment]
Shouldn't `[part="drawer"]` be using `var(--_vaadin-app-layout-drawer-offset-size)` as its width?

**Comment by jouni** [comment]
@anezthes, well, not really. The value of that custom property is set during the rendering and is measured from the actual width of the drawer. You could set the width of the drawer to `auto`, and let it be sized by its contents, and the custom property would match that width.
